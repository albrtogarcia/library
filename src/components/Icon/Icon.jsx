import PropTypes from 'prop-types'
import {
  IconArrowDown,
  IconArrowDownLeft,
  IconArrowDownRight,
  IconArrowLeft,
  IconArrowLeftRight,
  IconArrowRight,
  IconArrowUp,
  IconArrowUpDown,
  IconArrowUpLeft,
  IconArrowUpRight,
  IconBell,
  IconBriefcase,
  IconCalendar,
  IconCheck,
  IconChevronDown,
  IconChevronLeft,
  IconChevronRight,
  IconChevronsUpDwon,
  IconChevronUp,
  IconClipboardList,
  IconClose,
  IconCornerDownRight,
  IconDanger,
  IconDownload,
  IconEdit,
  Iconeraser,
  IconEuro,
  IconEye,
  IconFileCheck,
  IconFileDown,
  IconFileQuestion,
  IconFiles,
  IconFilter,
  IconGlasses,
  IconInfo,
  IconKey,
  IconListVideo,
  IconLogIn,
  IconLogOut,
  IconMail,
  IconMoreVertical,
  IconSidebar,
  IconSidebarClose,
  IconSidebarOpen,
  IconSortDesc,
  IconSuccess,
  IconTrash,
  IconUpload,
  IconUser,
  IconWarning
} from '../../assets/icon'
import './Icon.scss'

const iconTypes = {
  arrow_down: IconArrowDown,
  arrow_down_left: IconArrowDownLeft,
  arrow_down_right: IconArrowDownRight,
  arrow_left: IconArrowLeft,
  arrow_left_right: IconArrowLeftRight,
  arrow_right: IconArrowRight,
  arrow_up: IconArrowUp,
  arrow_up_down: IconArrowUpDown,
  arrow_up_left: IconArrowUpLeft,
  arrow_up_right: IconArrowUpRight,
  bell: IconBell,
  briefcase: IconBriefcase,
  calendar: IconCalendar,
  check: IconCheck,
  chevron_down: IconChevronDown,
  chevron_left: IconChevronLeft,
  chevron_right: IconChevronRight,
  chevrons_up_down: IconChevronsUpDwon,
  chevron_up: IconChevronUp,
  clipboard_list: IconClipboardList,
  close: IconClose,
  corner_down_right: IconCornerDownRight,
  danger: IconDanger,
  download: IconDownload,
  edit: IconEdit,
  eraser: Iconeraser,
  euro: IconEuro,
  eye: IconEye,
  file_check: IconFileCheck,
  file_down: IconFileDown,
  file_question: IconFileQuestion,
  files: IconFiles,
  filter: IconFilter,
  glasses: IconGlasses,
  info: IconInfo,
  key: IconKey,
  list_video: IconListVideo,
  log_in: IconLogIn,
  log_out: IconLogOut,
  mail: IconMail,
  more_vertical: IconMoreVertical,
  sidebar: IconSidebar,
  sidebar_close: IconSidebarClose,
  sidebar_open: IconSidebarOpen,
  sort_desc: IconSortDesc,
  success: IconSuccess,
  trash: IconTrash,
  upload: IconUpload,
  user: IconUser,
  warning: IconWarning
}

export const Icon = ({ name, ...props }) => {
  let Icon = iconTypes[name]
  return <Icon className="icon" {...props} />
}

Icon.propTypes = {
  name: PropTypes.string
}
