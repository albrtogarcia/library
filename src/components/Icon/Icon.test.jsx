import { describe, expect, it } from 'vitest'
import { Icon } from './Icon.jsx'

describe('Icon', () => {
  const props = {
    name: 'success'
  }

  it('matches snapshot', () => {
    const icon = Icon(props)
    expect(icon).toMatchSnapshot()
  })
})
