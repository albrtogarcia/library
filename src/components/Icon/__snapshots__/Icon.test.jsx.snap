// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon > matches snapshot 1`] = `
<SvgSuccess
  className="icon"
/>
`;
